zinc/src/main/scala/sbt/internal/inc/MixedAnalyzingCompiler.scala

```diff
@@ -417,12 +417,12 @@ object MixedAnalyzingCompiler {
     val absClasspath = classpath.map(toAbsolute(_))
     val cArgs =
       new CompilerArguments(compiler.scalaInstance, compiler.classpathOptions)
-    val searchClasspath: Seq[VirtualFile] = explicitBootClasspath(scalacOptions, converter) ++
+    val searchClasspath: Seq[VirtualFile] = tempJavacOutput ++ explicitBootClasspath(scalacOptions, converter) ++
```

Comment:
Same. It would be good to test shadowing of Java classes if it's currently not tested.